cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -828,8 +828,20 @@ def set_args(self, request, verbose=False):
         else:
             raise NotImplementedError
 
-        self.par_dic = args.to_dict()
-
+        self.par_dic = {}
+        for k, v in args.to_dict().items():
+            if k in ['catalog_selected_objects', 'selected_catalog']:
+                self.par_dic[k] = v
+                continue
+            try:
+                decoded = json.loads(v)
```

Comment:
it's a bit dangerous to rely on json parsing exception to determine if it should be parsed.

I guess it's hard to distinguish the intended type in here?

Reply:
`args` comes here from request, it's an ordered dict of strings. Need to parse them anyway to distinguish types.

If argument is actually a string, it's not a valid json because it isn't enclosed by `"` and we get the parsing exception.
If it's numeral, will parse successfully as int or float, not an object, then keep the previous behaviour.
If an object, decode.

We don't expect any other input here, but just in case, it won't parse, so keep previous behavior again.

Reply:
Right. I am concerned that if the json is somehow malformed where we expect it, we end up with a string which will lead later to possibly incomprehensive error. Maybe you can add a test for treating a failure like that?

Reply:
Hm, without any schema, the StructuredParameter accepts the string, may it be malformed json or anything. 
Which could be OK. Or we can add a restriction there, I'm not sure...

Reply:
So now, by default, StructuredParameter is required to really be structured. 
Added test with malformed structure